pkg/core/sync.go

```diff
@@ -219,6 +232,38 @@ func createProviderTables(ctx context.Context, db execution.QueryExecer, resourc
 	return diags
 }
 
+func safeToDropTable(ctx context.Context, db execution.QueryExecer, resName, table string) diag.Diagnostics {
+	rows, err := db.Query(ctx, `
+SELECT DISTINCT CONCAT(dependent_ns.nspname, '.', dependent_view.relname) as dependent_view
+FROM pg_depend
+JOIN pg_rewrite ON pg_depend.objid = pg_rewrite.oid
+JOIN pg_class as dependent_view ON pg_rewrite.ev_class = dependent_view.oid
+JOIN pg_class as source_table ON pg_depend.refobjid = source_table.oid
+JOIN pg_attribute ON pg_depend.refobjid = pg_attribute.attrelid AND pg_depend.refobjsubid = pg_attribute.attnum
+JOIN pg_namespace dependent_ns ON dependent_ns.oid = dependent_view.relnamespace
+JOIN pg_namespace source_ns ON source_ns.oid = source_table.relnamespace
+WHERE source_ns.nspname = current_schema() AND source_table.relname = $1 AND pg_attribute.attnum > 0 ORDER BY 1
+`, table)
+	if err != nil {
+		return diag.FromError(err, diag.DATABASE, diag.WithSummary("error checking dependent views"))
+	}
+
+	var dependentViews []string
+	if err := pgxscan.ScanAll(&dependentViews, rows); err != nil {
+		return diag.FromError(err, diag.DATABASE)
+	}
+
+	if len(dependentViews) > 0 {
+		return diag.Diagnostics{diag.NewBaseError(nil, diag.USER,
+			diag.WithSummary("One or more views depend on table %q, which is about to be dropped. Run with --force-drop to override this message which will drop all dependent views.", table),
```

Comment:
maybe capitalize `ALL`? because it will drop dependent views from ALL DROPPED tables